pallets/msa/src/lib.rs

```diff
@@ -1457,6 +1464,36 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Upload logo bytes for a approved image in `ApprovedLogos`
+		///
+		/// * [`Error::NoKeyExists`] - If there is not MSA for `origin`.
+		/// * [`Error::ProviderNotRegistered`] - If the provider is not registered.
+		/// * [`Error::InvalidCid`] - If the provided CID is invalid and not in approved logos.
+		/// * [`Error::LogoCidNotApproved`] - If the logo CID is not in the approved logos list.
+		///
+		#[pallet::call_index(24)]
+		#[pallet::weight(T::WeightInfo::upload_logo())]
+		pub fn upload_logo(
+			origin: OriginFor<T>,
+			logo_cid: BoundedVec<u8, T::MaxLogoCidSize>,
+			logo_bytes: BoundedVec<u8, T::MaxLogoSize>,
+		) -> DispatchResult {
+			let provider_key = ensure_signed(origin)?;
+			let provider_msa_id = Self::ensure_valid_msa_key(&provider_key)?;
+			ensure!(
+				Self::is_registered_provider(provider_msa_id),
+				Error::<T>::ProviderNotRegistered
+			);
+			ensure!(ApprovedLogos::<T>::contains_key(&logo_cid), Error::<T>::LogoCidNotApproved);
+
+			let input_cid_binary = Self::validate_cid(&logo_cid)?;
+			let computed_cid_binary =
+				compute_cid_v1(logo_bytes.as_slice()).ok_or(Error::<T>::InvalidLogoBytes)?;
+			ensure!(input_cid_binary == computed_cid_binary, Error::<T>::InvalidLogoBytes);
+			ApprovedLogos::<T>::insert(&logo_cid, logo_bytes);
+			Ok(())
```

Comment:
do we want to emit an event on successful upload? What about a generic ApplicationContextUpdated or ProviderUpdated event?